### PR TITLE
EmbedPanel: Add a postMessage to embedded panel

### PR DESCRIPTION
**Pull request by Mochi009**
What is this feature?

In Grafana, embed a panel is a massive feature but it's not possible to interact with it. With this implementation of a postMessage, the time for the update of a panel in an iframe is much shorter. 

Why do we need this feature?

This feature helps to minimize the time to update an embed panel using a postMessage instead of reloading all panel.

Who is this feature for?

All user that use grafana for the embedding can use it.

This functionality has been added within the files GrafanaRoute, SoloPanelPage, DashboardPanel, and PanelStateWrapper, which handle the creation of panels. In GrafanaRoute, the receiveMessage function takes a panelId as input, which is used to update the queryParams passed to SoloPanelPage, consequently updating the displayed panel. In PanelStateWrapper, on the other hand, the receiveMessage is designed to receive an array of key/value objects representing the variables to be updated for re-rendering the panel. All updates are performed using functions already present in Grafana.
This new functionality is helpfull when someone wants to update the variables the displayed panel uses. Instead of reloading all of its content, just the API for new data is triggered to display the new panel with new data.

### CODE REVIEW

**Comment by Mochi009** [comment]
I want to add lables abount changelog and milestone but i can't. Any advice? Thanks!

**Comment by aangelisc** [comment]
Hi @Mochi009, I've added the required label and the milestone. Please could you remove the changes to the `CHANGELOG` file?

**Comment by Mochi009** [comment]
Thanks @aangelisc. Can you help me with the other test that is blocked right now?

File: public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx [hunk 1]
```diff
@@ -235,6 +236,42 @@ export class PanelStateWrapper extends PureComponent<Props, State> {
 
     // Listen for live timer events
     liveTimer.listen(this);
+
+    // receive message from parent window
+    const receiveMessage = (event: any) => {
```
Comment (aangelisc) [inline public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx:241]: This should be typed correctly.

File: public/app/core/navigation/GrafanaRoute.tsx [hunk 2]
```diff
@@ -41,6 +43,21 @@ export function GrafanaRoute(props: Props) {
     navigationLogger('GrafanaRoute', false, 'Updated', props);
   });
 
+  // receive panelId from parent window
+  useEffect(() => {
+    const receiveMessage = (event: any) => {
```
Comment (aangelisc) [inline public/app/core/navigation/GrafanaRoute.tsx:48]: This should be typed correctly.

**Comment by aangelisc** [comment]
Hi @Mochi009,

Apologies for the delay. There are a couple failures preventing this from being mergeable.

- In `GrafanaRoute.tsx` and `PanelStateWrapper.tsx` you've add `any` types which our `betterer` configuration doesn't permit. Is it possible to type these correctly? I've commented the exact locations.
- The lint is falling, are you able to run `yarn lint:fix`
- Could you also merge main please?

Thanks!

**Comment by Mochi009** [comment]
Thanks @aangelisc. I fix problems with any type and lint. I also sync my repository with grafana main branch.

**Comment by baldm0mma** [comment]
Unblocked the Drone `continuous-integration/drone/pr` build step.

**Comment by Mochi009** [comment]
Hi @baldm0mma. I think every build you have to unlock it. Before it, can you check the other build and help me to pass all the test that failed? Thanks!

**Comment by baldm0mma** [comment]
@Mochi009 since you're not a member of the Grafana team, you don't have auth for interacting with enterprise content, so anything with `enterprise-clone` will fail, but it looks like you're having some linting issues still. Can you see the details for `continuous-integration/drone/pr`?

**Comment by Mochi009** [comment]
Hi @baldm0mma. I'm working on that error. I will update you when i make a fix for it. Thanks!

**Comment by Mochi009** [comment]
@baldm0mma i fix it. I think now it will pass all tests

**Comment by Mochi009** [comment]
@baldm0mma i fix lint bug but i have problems with betterer-frontend and verify-i18n
Can you help me? Thanks again!

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. Just letting you know I'm updating your PR's milestone to [11.2.x](https://github.com/grafana/grafana/milestone/529). If you need to see your changes in a previous version, please consider adding a backport label.

Thank you!

**Comment by aangelisc** [comment]
Pinging @grafana/dashboards-squad for review here 😊

**Pull request closed without merging**